### PR TITLE
Adds popups to maps

### LINKED ISSUE #756: Improve map interactivity: Allow users to select entities
## Context:

- Copy pattern [from consumer platform](https://www.planning.data.gov.uk/map/?dataset=article-4-direction-area#52.54505129189863,-2.126492382419201,11.449025421290857z).
- Instead of linking to the consumer entity page, just show the entity name (not linked to anywhere for now).

Image

- What do we need to consider from an accessibility standpoint?

## Acceptance criteria:

- Makes sure it works for dataset details page and check tool.
 
## Implementation details:

-

### PR DESCRIPTION
**Pull request by DilwoarH**
This allows maps to be clickable and information to be seen about each feature of a map.

FYI: This feature does not currently work on the check service as the service currently only sends WKT values and not features.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds popups to maps

## Related Tickets & Documents

- Ticket Link
- Related Issue #
- Closes #756 

## QA Instructions, Screenshots, Recordings

Screenshot 2025-01-04 at 7 00 03 pm
Screenshot 2025-01-04 at 7 00 09 pm

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _Please replace this line with details on why tests have not been included_
- [ ] I need help with writing tests

## QA sign off
- [ ] Code has been checked and approved
- [ ] Design has been checked and approved
- [ ] Product and business logic has been checked and proved

## Summary by CodeRabbit

- **New Features**
	- Enhanced map interaction with new popup functionality.
	- Added ability to display detailed feature information when clicking on the map.
	- Improved user experience with dynamic popup content and scrolling management.

- **Style**
	- Added new CSS styling for map popup list items.
	- Updated global stylesheet to include map component styles.

- **Tests**
	- Expanded test coverage for map functionalities, including edge cases and interactivity handling.

### CODE REVIEW

File: src/assets/js/map.js [hunk 1]
```diff
@@ -243,6 +314,35 @@ export const calculateBoundingBoxFromGeometries = (geometries) => {
   return [[minX, minY], [maxX, maxY]]
 }
 
+// Prevents scrolling of the page when the user triggers the wheel event on a div
+// while still allowing scrolling of any specified scrollable child elements.
+// Params:
+//  scrollableChildElements: an array of class names of potential scrollable elements
+const preventScroll = (scrollableChildElements = []) => {
+  return (e) => {
```
Comment (rosado) [inline src/assets/js/map.js:322]: FYI, You can reduce it to a single query by passing the selector `scrollableChildElements.join(', ')` to `Element.closest()`

**Review by rosado — DISMISSED** [review]
I wish `<template>` element was used in place of all the DOM wrangling code, but that's nitpick. Looks OK.

File: src/assets/js/map.js (hunk 1 above)
Comment (DilwoarH) [reply src/assets/js/map.js:322]: I kept it same as the one on the main website, all of this needs to be standardised.

**Comment by DilwoarH** [comment]
@CharliePatterson - are you happy for this to be merged?

**Comment by CharliePatterson** [comment]
LGTM 🙌

**Review by rosado — APPROVED** [review]
